### PR TITLE
Add custom training loop for transformer v2.

### PR DESCRIPTION
**Pull request by lintian06**
Add custom training loop and eval in Transformer V2.

- Tested with the following command for TF V1 and V2:
```
python official/transformer/v2/transformer_main.py \
  --data_dir=${DATA_DIR} \
  --model_dir=${MODEL_DIR} \
  --vocab_file=${VOCAB_FILE} \
  --param_set=${PARAM_SET} \
  --bleu_source=${BLEU_SOURCE} \
  --bleu_ref=${BLEU_REF} \
  --train_steps=10 \
  --steps_between_evals=5 \
  --batch_size=2048 \
  --mode=custom_train
```
- Tested with larger train_steps, which shows the decreasing loss.

### CODE REVIEW

File: official/transformer/v2/transformer.py [hunk 1]
```diff
@@ -85,13 +85,13 @@ def get_config(self):
         "params": self.params,
     }
 
-  def call(self, x, training):
+  def call(self, inputs, training):
```
Comment (lintian06) [inline official/transformer/v2/transformer.py]: `inputs` is a conventional parameter that handles nested tensors correctly.

**Comment by tfboyd** [comment]
I will leave this to @guptapriya   There is too much here I do not know enough about.

**Comment by guptapriya** [comment]
@rachellj218  is considering this as well, so will wait to review this until @lintian06 and @rachellj218 have discussed the best way to move forward.

**Comment by jaeyounkim** [comment]
I am closing this PR because we made many changes already.

**Pull request closed without merging**